app/code/Magento/Sitemap/Model/ItemProvider/CmsPage.php

```diff
@@ -32,38 +41,86 @@ class CmsPage implements ItemProviderInterface
+     */
+    private $pageFactory;
+
     /**
      * CmsPage constructor.
      *
-     * @param ConfigReaderInterface $configReader
-     * @param PageFactory $cmsPageFactory
-     * @param SitemapItemInterfaceFactory $itemFactory
+     * @param ConfigReaderInterface          $configReader
+     * @param PageFactory                    $cmsPageFactory
+     * @param SitemapItemInterfaceFactory    $itemFactory
+     * @param StoreManagerInterface          $storeManager
+     * @param ScopeConfigInterface           $scopeConfig
+     * @param \Magento\Cms\Model\PageFactory $pageFactory
      */
     public function __construct(
         ConfigReaderInterface $configReader,
         PageFactory $cmsPageFactory,
-        SitemapItemInterfaceFactory $itemFactory
+        SitemapItemInterfaceFactory $itemFactory,
+        StoreManagerInterface $storeManager,
+        ScopeConfigInterface $scopeConfig,
+        \Magento\Cms\Model\PageFactory $pageFactory
     ) {
         $this->cmsPageFactory = $cmsPageFactory;
         $this->itemFactory = $itemFactory;
         $this->configReader = $configReader;
+        $this->storeManager = $storeManager;
+        $this->scopeConfig = $scopeConfig;
+        $this->pageFactory = $pageFactory;
     }
 
     /**
-     * {@inheritdoc}
+     * @inheritdoc
      */
     public function getItems($storeId)
     {
         $collection = $this->cmsPageFactory->create()->getCollection($storeId);
-        $items = array_map(function ($item) use ($storeId) {
-            return $this->itemFactory->create([
-                'url' => $item->getUrl(),
-                'updatedAt' => $item->getUpdatedAt(),
-                'images' => $item->getImages(),
-                'priority' => $this->configReader->getPriority($storeId),
-                'changeFrequency' => $this->configReader->getChangeFrequency($storeId),
-            ]);
-        }, $collection);
+        $items = array_map(
+            function ($item) use ($storeId) {
+                return $this->itemFactory->create(
+                    [
+                    'url' => $item->getUrl(),
+                    'updatedAt' => $item->getUpdatedAt(),
+                    'images' => $item->getImages(),
+                    'priority' => $this->configReader->getPriority($storeId),
+                    'changeFrequency' => $this->configReader->getChangeFrequency($storeId),
+                    ]
+                );
+            },
+            $collection
+        );
+
+        $homePageIdentifierValue = $this->scopeConfig->getValue(
+            self::XML_PATH_HOMEPAGE_IDENTIFIER,
+            ScopeInterface::SCOPE_STORE
+        );
+
+        $homepage = $this->pageFactory->create()->load($homePageIdentifierValue, 'identifier');
+
+        /**
+         * Add configured homepage to sitemap using base_url as url
+         */
+        $items[] = $this->itemFactory->create(
+            [
+            'url' => $this->storeManager->getStore($storeId)->getBaseUrl(),
```

Comment:
I think it's better to use URL builder `Magento\Framework\UrlInterface::getUrl` method with an empty route parameter in order to properly render the homepage URL.